File: app/src/main/java/com/example/mg/masterdetail/data/ILoadItemsInteractor.java
package com.example.mg.masterdetail.data;

import com.example.mg.masterdetail.data.model.Weather10daysModel;
import com.example.mg.masterdetail.data.model.WeatherModel;

import io.reactivex.Single;

public interface ILoadItemsInteractor {
    Single<WeatherModel> weatherCityData();

    Single<Weather10daysModel> weather10DaysData();

}

File: app/src/main/java/com/example/mg/masterdetail/data/ILoadItemsInteractor.kt
package com.example.mg.masterdetail.data

import com.example.mg.masterdetail.data.model.WeatherDaysModel
import com.example.mg.masterdetail.data.model.WeatherModel

interface ILoadItemsInteractor {

    suspend fun weatherCityData(): Result<WeatherModel>

    suspend fun weatherDaysData(): Result<WeatherDaysModel>
}